### PR TITLE
mlir: bump llvm tag to 5380e3

### PR DESCRIPTION
**Pull request by ashay**
In addition to updating the llvm-project submodule, this patch also:

1. updates shape functions and tests so that `func` and `call`
   operations refer to the `func` dialect
2. avoid duplicate registration of dialects

### CODE REVIEW

**Comment by powderluv** [comment]
Will let others review but just wanted to say thanks for the solid contributions.

**Comment by ashay** [comment]
Thanks @powderluv!  I appreciate the welcoming community that you and others have created, so thanks to you, too!

**Comment by silvasean** [comment]
+1 Amazing work Ashay -- it is very much appreciated.

**Review by silvasean — APPROVED** [review]

**Comment by powderluv** [comment]
merging to resolve some downstream func.func issues

File: lib/Dialect/Torch/Transforms/ShapeLibrary.cpp [hunk 1]
```diff
@@ -28,15 +28,15 @@ StringRef mlir::torch::Torch::getShapeLibrary() {
 #pragma clang diagnostic ignored "-Woverlength-strings"
   constexpr StringLiteral shapeLib(R"mlir(
 module {
-  func @"__torch_mlir_shape_fn.aten.tanh"(%arg0: !torch.list<int>) -> !torch.list<int> {
-    %0 = call @__torch__.torch_mlir.dialects.torch.importer.jit_ir.build_tools.upstream_shape_helpers.unary(%arg0) : (!torch.list<int>) -> !torch.list<int>
+  func.func @"__torch_mlir_shape_fn.aten.tanh"(%arg0: !torch.list<int>) -> !torch.list<int> {
+    %0 = func.call @__torch__.torch_mlir.dialects.torch.importer.jit_ir.build_tools.upstream_shape_helpers.unary(%arg0) : (!torch.list<int>) -> !torch.list<int>
```
[the remark this answers is not in the document]
Comment (ashay) [reply lib/Dialect/Torch/Transforms/ShapeLibrary.cpp:32]: It's been a while but it looks like I must have manually replaced them (sed script).  If I rerun the script, it does revert the change.  Sorry, I'm now looking into the script that generates these lines.

**Pull request merged**